OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,245 @@
+        self._init_bars(emotion_size=emotion_size, emotion_images_path=emotion_images_path)
+
+    def _init_bars(self, emotion_size, emotion_images_path):
+        self.bars = {
+            "happy": {"bar": EmotionBar(emotion='happy', bar_size=self.bar_size,
+                                        emotion_size=emotion_size,
+                                        emotion_path=os.path.join(emotion_images_path, 'happy.png')), "count": 0},
+            "sad": {"bar": EmotionBar(emotion='sad', bar_size=self.bar_size, emotion_size=emotion_size,
+                              emotion_path=os.path.join(emotion_images_path, 'sad.png')), "count": 0},
+            "anger": {"bar": EmotionBar(emotion='anger', bar_size=self.bar_size, emotion_size=emotion_size,
+                                emotion_path=os.path.join(emotion_images_path, 'anger.png')), "count": 0},
+            "surprise": {"bar": EmotionBar(emotion='surprise', bar_size=self.bar_size, emotion_size=emotion_size,
+                                   emotion_path=os.path.join(emotion_images_path, 'surprise.png')), "count": 0},
+            "neutral": {"bar":EmotionBar(emotion='neutral', bar_size=self.bar_size, emotion_size=emotion_size,
+                                  emotion_path=os.path.join(emotion_images_path, 'neutral.png')), "count": 0}
+        }
+        self._fill_bars()
+
+    def _fill_bars(self):
+        for x in self.bars.values():
+            x["bar"].draw()
+
+    def _increment_count(self, emotion):
+        self.bars[emotion]["count"] += 1
+
+    def _relative_size_estimation(self, visualization_size, bar_amount: int = 5):
+        bar_section_size = visualization_size / bar_amount
+        bar_size = int(bar_section_size - 2 * bar_section_size * self.emotion_bar_offset_percent)
+        padding = int(bar_section_size - bar_size)
+        emotion_size = (int(bar_size / 3), int(bar_size / 3))
+        return bar_size, padding,  emotion_size
+
+    def draw_bars(self, vis_result):
+        y_offset = int(vis_result.shape[1] * self.y_offset_percent)
+        alpha_mask_overlay = self.overlay[:, :, 3] / 255.0
+        overlay_background = self.overlay[..., :3]
+        vis_result = overlay_image(vis_result, overlay_background, 0, 0, alpha_mask_overlay)
+        start_x, start_y = np.clip(vis_result.shape[1] - self.bar_size - int(self.padding / 2), 0, vis_result.shape[1]), \
+                           np.clip(vis_result.shape[0] - self.bar_size - y_offset, 0,
+                                   vis_result.shape[0])
+        for emotion, emotion_bar in self.bars.items():
+            percent = self.get_current_emotion_percent(emotion)
+            text = f'{emotion.upper()} - {percent}%'
+            vis_result = add_class_names_and_percents(vis_result, [start_x, int(vis_result.shape[0] - y_offset / 2)],
+                                                      text)
+            alpha_mask = emotion_bar["bar"].alpha_mask
```

Comment:
If you want to use an alpha mask from EmotionBar, it should be property. 

But better to avoid overlaying with alpha mask and think about another way to add images on each other

Reply:
Which one, for example?